presto-main/src/main/java/com/facebook/presto/operator/aggregation/arrayagg/SetUnionFunction.java

```diff
@@ -0,0 +1,162 @@
+import com.facebook.presto.metadata.BoundVariables;
+import com.facebook.presto.metadata.FunctionManager;
+import com.facebook.presto.metadata.SqlAggregationFunction;
+import com.facebook.presto.operator.aggregation.AccumulatorCompiler;
+import com.facebook.presto.operator.aggregation.AggregationMetadata;
+import com.facebook.presto.operator.aggregation.AggregationMetadata.AccumulatorStateDescriptor;
+import com.facebook.presto.operator.aggregation.AggregationMetadata.ParameterMetadata;
+import com.facebook.presto.operator.aggregation.AggregationUtils;
+import com.facebook.presto.operator.aggregation.GenericAccumulatorFactoryBinder;
+import com.facebook.presto.operator.aggregation.InternalAggregationFunction;
+import com.facebook.presto.operator.aggregation.SetOfValues;
+import com.facebook.presto.operator.aggregation.state.SetAggregationState;
+import com.facebook.presto.operator.aggregation.state.SetAggregationStateFactory;
+import com.facebook.presto.spi.function.AccumulatorState;
+import com.facebook.presto.spi.function.AccumulatorStateFactory;
+import com.facebook.presto.spi.function.AccumulatorStateSerializer;
+import com.google.common.collect.ImmutableList;
+
+import java.lang.invoke.MethodHandle;
+import java.util.List;
+
+import static com.facebook.presto.common.type.TypeSignature.parseTypeSignature;
+import static com.facebook.presto.operator.aggregation.AggregationMetadata.ParameterMetadata.ParameterType.BLOCK_INDEX;
+import static com.facebook.presto.operator.aggregation.AggregationMetadata.ParameterMetadata.ParameterType.NULLABLE_BLOCK_INPUT_CHANNEL;
+import static com.facebook.presto.operator.aggregation.AggregationMetadata.ParameterMetadata.ParameterType.STATE;
+import static com.facebook.presto.spi.function.Signature.typeVariable;
+import static com.facebook.presto.util.Reflection.methodHandle;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+
+public class SetUnionFunction
+        extends SqlAggregationFunction
+{
+    public static final SetUnionFunction SET_UNION = new SetUnionFunction();
+    private static final String NAME = "set_union";
+    private static final MethodHandle INPUT_FUNCTION = methodHandle(SetUnionFunction.class, "input", Type.class, Type.class, SetAggregationState.class, Block.class, int.class);
+    private static final MethodHandle COMBINE_FUNCTION = methodHandle(SetUnionFunction.class, "combine", SetAggregationState.class, SetAggregationState.class);
+    private static final MethodHandle OUTPUT_FUNCTION = methodHandle(SetUnionFunction.class, "output", SetAggregationState.class, BlockBuilder.class);
+
+    public SetUnionFunction()
+    {
+        super(NAME,
+                ImmutableList.of(typeVariable("T")),
+                ImmutableList.of(),
+                parseTypeSignature("array(T)"),
+                ImmutableList.of(parseTypeSignature("array(T)")));
+    }
+
+    @Override
+    public String getDescription()
+    {
+        return "return an array of unique values";
```

Comment:
Can we be a little more specific? This function definitely behaves differently from `set_agg` so they probably should not have exactly the same description?

Reply:
Done